src/ai/embedding/chunking/ai-chunking-fixed-size.ts

```diff
@@ -0,0 +1,41 @@
+// eslint-disable-next-line camelcase
+import { get_encoding } from 'tiktoken';
+
+/**
+ * Split a document into chunks based on fixed size of charecter and tokens
+ * @param root named parameters
+ * @param root.document the document to split into chunks
+ * @param root.separator a string such as \n or ' ' to seperate the chunk, default ' '
+ * @param root.tokenLimit maximum token length of the chunk, default 512
+ * @returns a list of chunks
+ */
+export function createLengthBasedChunks({
+  document,
+  separator = ' ',
```

Comment:
I need a comment here, (no need elsewhere) about why you chose the space as default.

It seems models now can handle \n or [space] fine

again, just a way to document stuff

additionally, if you think that the only separators needed are space or \n, you can specify this in typescript, to improve dev experience:

```ts
separator?: ' ' | '\n';
```